vector/src/main/java/im/vector/app/features/userdirectory/KnownUsersFragment.kt

```diff
@@ -107,6 +120,47 @@ class KnownUsersFragment @Inject constructor(
         }
     }
 
+    private fun setupAddByQrCodeView() {
+        val qrStartForActivityResult = registerStartForActivityResult { activityResult ->
+            if (activityResult.resultCode == Activity.RESULT_OK) {
+                val result = QrCodeScannerActivity.getResultText(activityResult.data)!!
+
+                // Latter condition to be replaced by URL(result).protocol == "matrix" when MSC2312 is complete
+                if (result.isValidUrl() && URL(result).host == "matrix.to") {
```

Comment:
You can use `PermalinkParser.parse(result)` it will simplify this code, and write something like
`val userId = (PermalinkParser.parse(result) as? PermalinkData.UserLink)?.userId`